crates/ursa/src/config.rs

```diff
@@ -1,26 +1,43 @@
+use resolve_path::PathResolveExt;
 use serde::{Deserialize, Serialize};
+use std::{
+    fs::{create_dir_all, File},
+    io::{Read, Result, Write},
+    path::{Path, PathBuf},
+    process,
+};
+use tracing::{error, info};
 use ursa_index_provider::config::ProviderConfig;
 use ursa_metrics::config::MetricsServiceConfig;
 use ursa_network::NetworkConfig;
 use ursa_rpc_server::config::ServerConfig;
 
-use std::{
-    fs::{create_dir_all, File},
-    io::{Result, Write},
-    path::PathBuf,
-};
-
 pub const DEFAULT_CONFIG_PATH_STR: &str = ".ursa/config.toml";
 
-pub fn load_config(path: &PathBuf) -> Result<()> {
+pub fn load_config(path: &PathBuf) -> Result<UrsaConfig> {
+    info!("Loading config from: {:?}", path);
     if !path.exists() {
         let ursa_config = UrsaConfig::default();
         let toml = toml::to_string(&ursa_config).unwrap();
         create_dir_all(path.parent().unwrap())?;
         let mut file = File::create(path)?;
-        return file.write_all(toml.as_bytes());
+        file.write_all(toml.as_bytes())?;
+        Ok(ursa_config)
+    } else {
+        // Read from config file
+        let toml = read_file_to_string(path)?;
+        // Parse and return the configuration file
+        let mut config: UrsaConfig = toml::from_str(&toml)?;
+        // parse relative and home directory paths
+        config.network_config.keystore_path =
+            config.network_config.keystore_path.resolve().to_path_buf();
+        config.network_config.database_path =
+            config.network_config.database_path.resolve().to_path_buf();
+        config.provider_config.database_path =
+            config.provider_config.database_path.resolve().to_path_buf();
+
+        Ok(config)
```

Comment:
I would try to maintain `load_config` to be less verbose. We don't need to parse each individual field separately

Reply:
moved the resolve calls to where the config values are instead of all at once